sktime/networks/base.py

```diff
@@ -25,3 +28,154 @@ def build_network(self, input_shape, **kwargs):
         output_layer : a keras layer
         """
         ...
+
+
+class BaseDeepNetworkPyTorch(BaseForecaster, ABC):
+    """Abstract base class for deep learning networks using torch.nn."""
+
+    _tags = {
+        "python_dependencies": "torch",
+    }
+
+    def __init__(self):
+        super().__init__()
+
+    def _fit(self, y, X, fh):
+        """Fit the network.
+
+        Changes to state:
+            writes to self._network.state_dict
+
+        Parameters
+        ----------
+        X : iterable-style or map-style dataset
+            see (https://pytorch.org/docs/stable/data.html) for more information
+        """
+        dataloader = self.build_pytorch_train_dataloader(y)
```

Comment:
I would propose that the creation of the dataloader considers the passed fh. This would also enable the user to create a forecaster that predicts the next 4 to 8 values instead of having to predict all the next 8 values. This would probably also solve the issue with your test and the forecast that does not fit to the fh.